qiita_plugins/biom_type/biom_type_plugin/tests/test_summary.py

```diff
@@ -0,0 +1,603 @@
+        # Create a biom table
+        fd, self.biom_fp = mkstemp(suffix=".biom")
+        close(fd)
+        data = np.asarray([[0, 0, 1], [1, 3, 42]])
+        table = Table(data, ['O1', 'O2'], ['1.S1', '1.S2', '1.S3'])
+        with biom_open(self.biom_fp, 'w') as f:
+            table.to_hdf5(f, "Test")
+        self.out_dir = mkdtemp()
+        self.artifact_id = 4
+        self.parameters = {'input_data': self.artifact_id}
+
+        self._clean_up_files = [self.biom_fp, self.out_dir]
+
+    def tearDown(self):
+        for fp in self._clean_up_files:
+            if exists(fp):
+                if isdir(fp):
+                    rmtree(fp)
+                else:
+                    remove(fp)
+
+    @httpretty.activate
+    def test_generate_html_summary(self):
+        httpretty_url = ("https://test_server.com/qiita_db/artifacts/"
+                         "%s/filepaths/" % self.artifact_id)
+        httpretty.register_uri(
+            httpretty.GET, httpretty_url,
+            body=('{"success": true, "error": "", '
+                  '"filepaths": [["%s", "biom"]]}' % self.biom_fp))
+        httpretty.register_uri(
+            httpretty.PATCH, httpretty_url,
+            body='{"success": true, "error": ""}')
+        obs, html = generate_html_summary(self.qclient, 'job-id',
+                                          self.parameters,
+                                          self.out_dir, True)
+
+        # asserting reply
+        self.assertItemsEqual(obs, {"success": True, "error": ""})
+
+        # asserting content of html
+        self.assertItemsEqual(html, EXT_HTML)
+
+    @httpretty.activate
+    def test_generate_html_summary_error(self):
+        httpretty_url = ("https://test_server.com/qiita_db/artifacts/"
+                         "%s/filepaths/" % self.artifact_id)
+        httpretty.register_uri(
+            httpretty.GET, httpretty_url,
+            body=('{"success": true, "error": "", '
+                  '"filepaths": [["%s", "biom"]]}' % self.biom_fp))
+        with self.assertRaises(ValueError):
+            generate_html_summary(self.qclient, 'job-id', self.parameters,
+                                  self.out_dir, True)
+
+    def test_count_summary(self):
+        biom = load_table(self.biom_fp)
+        obs = count_summary(biom)
+        exp_summary = {'Mean count': 15.666666666666666, 'Minimum count': 1.0,
+                       'Maximum count': 43.0, 'Median count': 3.0}
+        exp_counts = [1.0, 3.0, 43.0]
+        self. assertItemsEqual(obs[0], exp_summary)
+        self. assertItemsEqual(obs[1], exp_counts)
+
+
+EXT_HTML = [
+    '<b>Number of samples:</b> 3<br/>',
+    '<b>Number of features:</b> 2<br/>',
+    '<b>Minimum count:</b> 1<br/>',
+    '<b>Maximum count:</b> 43<br/>',
+    '<b>Median count:</b> 3<br/>',
+    '<b>Mean count:</b> 15<br/>',
+    '<br/><hr/><br/>',
+    ('<img src = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAyAAAAImCAYAA'
```

Comment:
We really don't need to have the image data, if we want to test the correctness of the figure, we should return the figure object itself and check its attributes.